js/index.js

```diff
@@ -1,98 +1,101 @@
-import { saveAs } from 'file-saver';
-
-import("../pkg/index.js")
-  .then(wasm => {
-    let Source = wasm.Source;
-
-    let member = {
-      inputEl: null,
-      dropEl: null,
-      infoEl: null
-    };
-
-    (function main() {
-      member = { ...initElements() };
-      checkFileApiSupport();
-    })();
-
-    function initElements() {
-      const containerEl = document.createElement("div");
-      containerEl.innerHTML = `<input type="file"/><br>
+import { saveAs } from 'file-saver'
+
+let member = {
+  inputEl: null,
+  dropEl: null,
+  infoEl: null,
+  WasmSource: null,
+}
+
+;(async function main() {
+  member = { ...initElements(), WasmSource: await getWasmSource() }
```

Comment:
great stuff! That's the syntax I've been looking for. Too bad it clashes with what I was trying todo. Merging yours.